save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/ManageGitCredentialsCard.kt

```diff
@@ -0,0 +1,268 @@
+                                else -> faHome
+                            },
+                            classes = "h-75 w-75"
+                        )
+                    }
+                    div {
+                        className = ClassName("col-7 text-left align-self-center pl-0")
+                        +url
+                    }
+                }
+                div {
+                    className = ClassName("col-5 align-self-right d-flex align-items-center justify-content-end")
+                    button {
+                        className = ClassName("btn col-2 align-items-center mr-2")
+                        fontAwesomeIcon(icon = faEdit)
+                        id = "edit-git-credential-$index"
+                        onClick = {
+                            setGitCredentialToUpsert(gitCredential)
+                            setGitWindowOpened(true)
+                        }
+                    }
+                    button {
+                        className = ClassName("btn col-2 align-items-center mr-2")
+                        fontAwesomeIcon(icon = faTimesCircle)
+                        id = "remove-git-credential-$index"
+                        onClick = {
+                            setGitCredentialToDelete(gitCredential)
+                            setConfirmDeleteGitCredentialWindowOpened(true)
+                        }
+                    }
+                    hidden = !canModify
+                }
+            }
+        }
+        @Suppress("EMPTY_BLOCK_STRUCTURE_ERROR")
+        hr {}
+        div {
+            className = ClassName("row d-flex justify-content-center")
+            div {
+                className = ClassName("col-3 d-sm-flex align-items-center justify-content-center")
+                button {
+                    type = ButtonType.button
+                    className = ClassName("btn btn-sm btn-primary")
+                    onClick = {
+                        setGitCredentialToUpsert(null)
+                        setGitWindowOpened(true)
+                    }
+                    +"Add new"
+                }
+            }
+        }
+    }
+
+    runOnlyOnFirstRender {
+        fetchGitCredentialsRequest()
+    }
+}
+
+@Suppress("TYPE_ALIAS")
+private fun prepareFetchGitCredentials(organizationName: String): RequestWithDependency<List<GitDto>> {
+    val (gitCredentials, setGitCredentials) = useState(emptyList<GitDto>())
+    val fetchGitCredentialsRequest = useRequest {
+        get(
+            "$apiUrl/organizations/$organizationName/list-git",
+            headers = headers,
+            loadingHandler = ::loadingHandler,
+        )
+            .unsafeMap {
+                it.decodeFromJsonString<List<GitDto>>()
+            }.let {
+                setGitCredentials(it)
+            }
+    }
+    return Triple(gitCredentials, setGitCredentials, fetchGitCredentialsRequest)
+}
+
+private fun prepareUpsertGitCredential(
```

Comment:
need to check that we add gitUrl which is not added yet

Reply:
agreed to leave current logic -- if we add a new git credentials with existed url -- it updates existed value